test/cli/install/bun-add.test.ts

```diff
@@ -2324,3 +2324,104 @@ it("should add multiple dependencies specified on command line", async () => {
   });
   await access(join(package_dir, "bun.lockb"));
 });
+
+it("should install tarball with tarball dependencies", async () => {
+  // This test verifies that tarballs containing dependencies that are also tarballs
+  // can be installed correctly. Regression test for URL corruption bug.
+
+  // Need to update the parent tarball to use the actual server URL
+  const parentPkgJson = {
+    name: "tarball-parent",
+    version: "0.0.1",
+    dependencies: {
+      "tarball-child": "http://localhost:6789/tarball-child-0.0.1.tgz",
```

Comment:
this looks like too much hard coding. is it a guarantee that it'll be running on that port or is this not actually being installed?